server/utils/forms/inputs/twelveHourClockTimeInput.test.ts

```diff
@@ -0,0 +1,200 @@
+
+      expect(result.error).toEqual({
+        errors: [
+          {
+            errorSummaryLinkedField: 'alarm-time-part-of-day',
+            formFields: ['alarm-time-part-of-day'],
+            message: 'Select whether the alarm time is AM or PM',
+          },
+        ],
+      })
+    })
+
+    it('returns an error when part of day is neither “am” nor “pm”', async () => {
+      const request = TestUtils.createRequest({
+        'alarm-time-hour': '1',
+        'alarm-time-minute': '09',
+        'alarm-time-part-of-day': 'rabbit',
+      })
+
+      const result = await new TwelveHourClockTimeInput(request, 'alarm-time', messages).validate()
+
+      expect(result.error).toEqual({
+        errors: [
+          {
+            errorSummaryLinkedField: 'alarm-time-part-of-day',
+            formFields: ['alarm-time-part-of-day'],
+            message: 'The alarm time must be a real time',
+          },
+        ],
+      })
+    })
+
+    it('returns an error when a field is just whitespace', async () => {
+      const request = TestUtils.createRequest({
+        'alarm-time-hour': '1',
+        'alarm-time-minute': '     ',
+        'alarm-time-part-of-day': 'pm',
+      })
+
+      const result = await new TwelveHourClockTimeInput(request, 'alarm-time', messages).validate()
+
+      expect(result.error).toEqual({
+        errors: [
+          {
+            errorSummaryLinkedField: 'alarm-time-minute',
+            formFields: ['alarm-time-minute'],
+            message: 'The alarm time must include a minute',
+          },
+        ],
+      })
+    })
+
+    it('returns an error when a field is non-numeric', async () => {
+      const request = TestUtils.createRequest({
+        'alarm-time-hour': '1',
+        'alarm-time-minute': 'hello',
+        'alarm-time-part-of-day': 'pm',
+      })
+
+      const result = await new TwelveHourClockTimeInput(request, 'alarm-time', messages).validate()
+
+      expect(result.error).toEqual({
+        errors: [
+          {
+            errorSummaryLinkedField: 'alarm-time-minute',
+            formFields: ['alarm-time-minute'],
+            message: 'The alarm time must be a real time',
+          },
+        ],
+      })
+    })
+
+    it('returns an error when a field is not an integer', async () => {
+      const request = TestUtils.createRequest({
+        'alarm-time-hour': '1',
+        'alarm-time-minute': '5.1',
+        'alarm-time-part-of-day': 'pm',
+      })
+
+      const result = await new TwelveHourClockTimeInput(request, 'alarm-time', messages).validate()
+
+      expect(result.error).toEqual({
+        errors: [
+          {
+            errorSummaryLinkedField: 'alarm-time-minute',
+            formFields: ['alarm-time-minute'],
+            message: 'The alarm time must be a real time',
+          },
+        ],
+      })
+    })
+
+    it('returns an error when the time specified is outside the boundaries of a 12-hour time', async () => {
```

Comment:
this one's interesting. almost tempting to correct 24 hour format hours isn't it. but this is good as is.